### PR TITLE
[Storage] Migrate `test_share` to test proxy (file-share)

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR migrates `test_share.py` and `test_share_async.py` to test proxy 😄

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/tests/test_share.py [hunk 1]
```diff
@@ -1,42 +1,41 @@
-# coding: utf-8
-
 # -------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
 import os
+import requests
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/tests/test_share.py:7]: nit: `requets` is 3rd party library and therefore was in the correct spot previously :)

File: sdk/storage/azure-storage-file-share/tests/test_share.py [hunk 2]
```diff
@@ -133,44 +144,52 @@ def test_create_snapshot_with_metadata(self, storage_account_name, storage_accou
...
-        with self.assertRaises(HttpResponseError):
+        with pytest.raises(HttpResponseError):
             share.delete_share()
 
         deleted = share.delete_share(delete_snapshots=True)
-        self.assertIsNone(deleted)
+        assert deleted is None
         self._delete_shares()
 
     @pytest.mark.playback_test_only
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/tests/test_share.py:174]: I assume we didn't have a file-share soft-delete account int he ARM template?

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/tests/test_share.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-share/tests/test_share.py]: Drats. Fixed 😄

File: sdk/storage/azure-storage-file-share/tests/test_share.py (hunk 2 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-share/tests/test_share.py:174]: Correct. I tried (albeit, for only ~5 minutes before giving up) to do the pattern-matching necessary but ended up deciding it wasn't worth the effort for 1 test (and it wasn't already nicely hooked up and just not exposed like in `datalake` 😢)

**Pull request merged**